### PR TITLE
feat(firebase_ui_localizations): Add Finnish language localization labels

### PR DESCRIPTION
**Pull request by launis**
## Description

This PR adds new language Finnish. The translations are checked but not reviewed by third party.

The melos run analyze noticed error on the original main and these continued to be the same when I did with my changes,
## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for _all_ changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ x] I updated/added relevant documentation (doc comments with `///`).
- [ x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ x] All unit tests pass (`melos run test:unit:all` doesn't fail).
- [ x] I read and followed the [Flutter Style Guide].
- [ x] I signed the [CLA].
- [ x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [ x] No, this is _not_ a breaking change.

[issue database]: https://github.com/firebase/FirebaseUI-Flutter/issues
[Contributor Guide]: https://github.com/firebase/FirebaseUI-Flutter/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by launis** [comment]
It looks like you've already signed this CLA. If you'd like to edit your contact information, you may do so below. 

git config user.email prints
risto.launis@gmail.com

and thats my name in the Contributor License Agreement  too, 

I have unchecked Keep my email addresses private in the github settings.

Something, that I do not understand

File: packages/firebase_ui_localizations/lib/src/get_localization.dart [hunk 1]
```diff
@@ -11,6 +11,7 @@ final Set<String> kSupportedLanguages = {
   'de', // German
   'en', // English
   'es', // Spanish Castilian
+  'fi', // Finnish
```
Comment (russellwheatley) [inline packages/firebase_ui_localizations/lib/src/get_localization.dart:14]: Finnish language needs adding to the switch statement here: https://github.com/firebase/FirebaseUI-Flutter/blob/main/packages/firebase_ui_localizations/lib/src/get_localization.dart#L57

**Review by russellwheatley — CHANGES_REQUESTED** [review]
Thanks for the PR! Could you also get a Finnish language speaker to also approve the PR and I will get this merged. Thanks 😄

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+  "@phoneNumberInvalidErrorText": {
+    "description": "Used as an error text when PhoneInput contains an invalid phone number.",
+    "placeholders": {}
+  },
+  "phoneNumberIsRequiredErrorText": "Puhelinnumero vaaditaan",
+  "@phoneNumberIsRequiredErrorText": {
+    "description": "Used as an error text when PhoneInput is empty.",
+    "placeholders": {}
+  },
+  "phoneVerificationViewTitleText": "Anna Puhelinnumerosi",
```
Comment (TimoPieti) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb:243]: No need to capitalize first letter of `Puhelinnumerosi`.

**Review by TimoPieti — COMMENTED** [review]
Looks good as a Finnish speaking user. Just a couple of observations.

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+  "cancelButtonLabel": "Peru",
+  "@cancelButtonLabel": {
+    "description": "Cancel button label",
+    "placeholders": {}
+  },
+  "unlinkProviderAlertMessage": "Haluatko varmasti poistaa tåmån kirjautmistavan linkityksen",
+  "@unlinkProviderAlertMessage": {
+    "description": "Text that is shown as a message of the AlertDialog confirming provider unlink",
+    "placeholders": {}
+  },
```
Comment (jpsiitonen) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb:511]: This doesn't look a valid translation - typos.
Probably should be something like: "Haluatko varmasti poistaa tämän kirjautumistavan linkityksen?"

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+  "@confirmDeleteAccountButtonLabel": {
+    "description": "Confirm delete account button label",
+    "placeholders": {}
+  },
+  "sendVerificationEmailLabel": "Lähetä vshvistussähköpsti",
+  "@sendVerificationEmailLabel": {
+    "description": "Used as a button label to send a verification email",
+    "placeholders": {}
+  }
+}
```
Comment (jpsiitonen) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb:537]: typo; should be like "Lähetä vahvistussähköposti"

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb [hunk 5]
```diff
@@ -0,0 +1,537 @@
...
+  "confirmUnlinkButtonLabel": "Poista linkitys",
+  "@confirmUnlinkButtonLabel": {
+    "description": "Unlink confirmation button label",
+    "placeholders": {}
+  },
+  "cancelButtonLabel": "Peru",
+  "@cancelButtonLabel": {
+    "description": "Cancel button label",
+    "placeholders": {}
+  },
```
Comment (jpsiitonen) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb:506]: in this case, i would probably consider term "Peruuta"

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+  "invalidVerificationCodeErrorText": "Antamasi koodi on virheellinen. Yritä uudelleen.",
+  "@invalidVerificationCodeErrorText": {
+    "description": "Error text indicating that entered SMS code is invalid",
+    "placeholders": {}
+  },
+  "ulinkProviderAlertTitle": "Poista kirjautmistavan linkitys",
+  "@ulinkProviderAlertTitle": {
+    "description": "Title that is shown in AlertDialog asking for provider unlink confirmation",
+    "placeholders": {}
+  },
```
Comment (jpsiitonen) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb:496]: contains typo, should be "Poista **kirjautumistavan** linkitys"

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb [hunk 7]
```diff
@@ -0,0 +1,537 @@
...
+  "signInWithAppleButtonText": "Kirjaudu Apple ID:llä",
+  "@signInWithAppleButtonText": {
+    "description": "Used as a label of the OAuthProviderButton for Apple provider.",
+    "placeholders": {}
+  },
+  "signInWithEmailLinkSentText": "Olemme lähettäneet sinulle kirjautumislinkin sisäältävän sähköpostiviestin. Tarkista sähköpostisi ja kirjaudu sisään linkin avulla.",
+  "@signInWithEmailLinkSentText": {
+    "description": "Indicates that email with a sign in link was sent.",
+    "placeholders": {}
+  },
```
Comment (jpsiitonen) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb:321]: typo in word "sisäältävän", should be "sisältävän"

Comment (launis) [reply packages/firebase_ui_localizations/lib/l10n/firebase_ui_fi.arb:321]: Thanks for all the comments. A person becomes so blind to what he himself has written, that's why he doesn't even see obvious mistakes. That "null" term is an excellent question. It just doesn't have a good Finnish counterpart.

In the next step I will do a new commit with corrected files.

File: packages/firebase_ui_localizations/lib/src/get_localization.dart (hunk 1 above)
Comment (launis) [reply packages/firebase_ui_localizations/lib/src/get_localization.dart:14]: I made the addition and created new commit. Perhaps a mention of the necessary edits to all needed files could be added to Firebase UI Localizations README.

**Review by jpsiitonen — APPROVED** [review]
Checked latest changes to translations. LGTM

**Review by TimoPieti — APPROVED** [review]
LGTM for me too.

**Comment by launis** [comment]
> Thanks for the PR! Could you also get a Finnish language speaker to also approve the PR and I will get this merged. Thanks 😄

As you see, that is done now.

**Comment by launis** [comment]
Reopened, can be closed when code changes are committed to main.

**Review by russellwheatley — APPROVED** [review]

**Pull request merged**